src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheck.java

```diff
@@ -139,7 +139,7 @@ public final void setIllegalPkgs(String... from) {
         illegalPkgs = from.clone();
         illegalPkgsRegexps.clear();
         for (String illegalPkg : illegalPkgs) {
-            illegalPkgsRegexps.add(CommonUtil.createPattern("^" + illegalPkg + "\\..*"));
+            illegalPkgsRegexps.add(CommonUtil.createPattern("^(static|" + illegalPkg + ")\\..*"));
```

Comment:
This is not much different from https://github.com/checkstyle/checkstyle/pull/13960#discussion_r1377565209 . This now looks like it can ignore the illegal package definition completely for the word `static`. Even if it can't be valid java to show that issue, this still doesn't seem right.